torch/testing/_internal/common_methods_invocations.py

```diff
@@ -358,6 +358,28 @@ def sample_inputs_tensor_split(op_info, device, dtype, requires_grad):
                         args=(torch.tensor([1, 2, 3]),),
                         kwargs=dict(dim=1)),)
 
+def sample_inputs_linalg_matrix_power(op_info, device, dtype, requires_grad):
+    test_sizes = [
+        (1, 1),
+        (S, S),
+        (0, S, S),
+        (S, S, S),
+    ]
+
+    inputs = []
+
+    for size in test_sizes:
+        for n in torch.arange(0, 5):
+            t = make_tensor(size, device, dtype, requires_grad=requires_grad)
+            inputs.append(SampleInput(t, args=(n,)))
+
+    for n in torch.arange(-4, 0):
+        t = random_fullrank_matrix_distinct_singular_value(S, S, device=device, dtype=dtype)
```

Comment:
I suggest to use 2 instead of S for batches.
```suggestion
        t = random_fullrank_matrix_distinct_singular_value(S, 2, device=device, dtype=dtype)
```